libtsuba/src/FileView.cpp

```diff
@@ -461,8 +461,8 @@ FileView::Resolve(int64_t start, int64_t size) {
         fetch->last_page >= page_number(start)) {
       // Complete the remaining work if there is some
       if (fetch->work.valid()) {
-        if (auto res = fetch->work.get(); !res) {
-          return res.error();
+        if (auto err = fetch->work.get(); err) {
+          return err.value();
```

Comment:
I'm gonna fight to not use `optional` like this. This is backwards where the "value" is an error and the "empty" is sucess. This means that `return err.value` is propagating an error which is super weird. `.value` shouldn't be an error except in code that is ONLY handling error codes. Can we create a real result type as discussed above?